### ISSUE TITLE
Nested arrays not mapped right - only first item is mapped

### ISSUE THREAD

**Issue by shay1e** [expressmapper/ExpressMapper#56]
Hi,

This are the models I use for testing:
```
//Source:
public class Source
{
	public SourceArrayContainer ArrayContainer;
}
public class SourceArrayContainer
{
	public SourceNestedClass[] nestedArray1;
}
public class SourceNestedClass
{
	public string SourceNestedString;
	public int SourceNestedInt;
	public string sameNameNestedString;
}
//Target:
public class Target
{
	public TargetArrayContainer ArrayContainer;
}
public class TargetArrayContainer
{
	public TargetNestedClass[] NestedArray1;
}
public class TargetNestedClass
{
	public string TargetNestedString;
	public int TargetNestedInt;
	public string SameNameNestedString;
}
```

And the test:
```
Mapper.Reset();
Mapper.Register<SourceNestedClass, TargetNestedClass>()
	.Member(p => p.TargetNestedInt, p => p.SourceNestedInt)
	.Member(p => p.TargetNestedString, p => p.SourceNestedString);
Mapper.Register<SourceNestedClass[], TargetNestedClass[]>();
Mapper.Register<SourceArrayContainer, TargetArrayContainer>();  //ProcessAutoProperties() should pickup the array field
Mapper.Register<Source, Target>();

var source = new Source()
{
	ArrayContainer = new SourceArrayContainer()
	{
		nestedArray1 = new SourceNestedClass[]
		{
			new SourceNestedClass() {SourceNestedInt = 1},
			new SourceNestedClass(){SourceNestedString = "x", sameNameNestedString = "y"}
		}
	}
};
var res = Mapper.Map(source, new Target());
```

Assuming my mapping configurations are correct (?), I would expect `res.ArrayContainer.NestedArray1` to have 2 items, correlates to `source.ArrayContainer.NestedArray1`. However, both items in `res.ArrayContainer.NestedArray1` are actually a mapping from the same object - `source.ArrayContainer.NestedArray1[0]`. Am I doing something wrong?

BTW, based on benchmarks we performed in our team, ExpressMapper won some few other mappers big time! :D

**Comment by shay1e** [comment]
Note:

When mapping `SourceArrayContainer` to `TargetArrayContainer` directly, without `Source` and `Target` as wrappers, the mapping works fine.

**Comment by anisimovyuriy** [comment]
Dear shay1e,

Thanks a lot for your valuable input and support!
There are couple of things that could be improved and fixed:
1) ```Mapper.Register<SourceNestedClass[], TargetNestedClass[]>();``` - it is not valid. You can register only classes - not collection types. Actually I'm going to create a ticket to produce an error if it is the case. - Thanks. <a href="https://github.com/expressmapper/ExpressMapper/issues/57">Here is the ticket!</a>
2) ```var res = Mapper.Map(source, new Target());``` - here you don't need to create an instance of "Target" in your case - it will be created automatically. When you specify existing destination parameter - another algorithm will be used that tries to reuse existing objects (Destination) rather than creating new ones. But anyway either it is with destination or just a source it should work in both cases. - Thanks! <a href="https://github.com/expressmapper/ExpressMapper/issues/58">Here is the ticket!</a>
3) 
```Mapper.Register<SourceNestedClass, TargetNestedClass>()
    .Member(p => p.TargetNestedInt, p => p.SourceNestedInt)
    .Member(p => p.TargetNestedString, p => p.SourceNestedString);```
Using just fields not properties in custom member mapping ```.Member``` - throws exception. <a href="https://github.com/expressmapper/ExpressMapper/issues/59">Here is the ticket!</a>

Thanks a lot once again shay1e for improving Expressmapper.
Your solution will work if you do the following modifications:
1) Using properties instead of fields - next release I'll make support for custom ```.Member``` fields.

```//Source:
public class Source
{
    public SourceArrayContainer ArrayContainer {get;set;}
}
public class SourceArrayContainer
{
    public SourceNestedClass[] nestedArray1 {get;set;}
}
public class SourceNestedClass
{
    public string SourceNestedString {get;set;}
    public int SourceNestedInt {get;set;}
    public string sameNameNestedString {get;set;}
}
//Target:
public class Target
{
    public TargetArrayContainer ArrayContainer {get;set;}
}
public class TargetArrayContainer
{
    public TargetNestedClass[] NestedArray1 {get;set;}
}
public class TargetNestedClass
{
    public string TargetNestedString {get;set;}
    public int TargetNestedInt {get;set;}
    public string SameNameNestedString {get;set;}
}
```

2) Removing registration of collection types and use ```.Map<TSrc,TDest>(TSrc source)``` - in your case

```Mapper.Reset();
Mapper.Register<SourceNestedClass, TargetNestedClass>()
    .Member(p => p.TargetNestedInt, p => p.SourceNestedInt)
    .Member(p => p.TargetNestedString, p => p.SourceNestedString);
Mapper.Register<SourceArrayContainer, TargetArrayContainer>();  
//ProcessAutoProperties() should pickup the array field
Mapper.Register<Source, Target>();

var source = new Source()
{
    ArrayContainer = new SourceArrayContainer()
    {
        nestedArray1 = new SourceNestedClass[]
        {
            new SourceNestedClass() {SourceNestedInt = 1},
            new SourceNestedClass(){SourceNestedString = "x", sameNameNestedString = "y"}
        }
    }
};
var res = Mapper.Map<Source, Target>(source);
```

**Comment by shay1e** [comment]
@anisimovyuriy Thank you so much for the detailed answer! I'm glad to hear this input is helpful :)
Actually, I didn't understand why fields are not supported so I just added it to `MemberConfiguration<T, TN>.Member()`. Please tell me, is there a special reason for that? I'm actually working on a wrapper for ExpressMapper, but I had to inject some code to the source code itself - and I would like to avoid that as much as possible.

**Comment by anisimovyuriy** [comment]
Dear shay1e,

It was a bug and you found it and it is fixed :)
https://github.com/expressmapper/ExpressMapper/issues/59
It is included for next release 1.7

Greatfully yours,
Yuriy

**Comment by anisimovyuriy** [comment]
Fixed and will be included in 1.7 release
